base.py

```diff
@@ -151,6 +162,9 @@ def run(self):
             
         except (NoSuchWindowException, WebDriverException):
             print('\nThe process was stopped, because the browser was closed.')
+        
+        except ConnectionError:
+            print('Failed to connect')
```

Comment:
It is a duplication of `requests.ConnectionError`, isn't it?
If not take your attention that exception text should be started with `\n` and finished with the dot.